src/consumer/message_bus.rs

```diff
@@ -35,4 +40,20 @@ pub trait IncomingMessage: Send + Sync + 'static {
 
     /// Reject message and dequeue
     async fn reject(&self) -> Result<(), Self::Error>;
+
+    async fn extract<C, M>(&self, codec: C) -> Result<M, MessageConvertError>
```

Comment:
I think this function should live outside of this trait - it shouldn't be overridden by `IncomingMessage` implementation imo